scipy/stats/_morestats.py

```diff
@@ -955,28 +978,39 @@ def boxcox_llf(lmb, data):
     >>> plt.show()
 
     """
+    # _axis_nan_policy decorator does not currently support these for non-NumPy arrays
+    kwargs = {}
+    if keepdims != False:
+        kwargs[keepdims] = keepdims
+    if nan_policy != 'propagate':
+        kwargs[nan_policy] = nan_policy
+    return _boxcox_llf(data, lmb=lmb, axis=axis, **kwargs)
+
+
+@_axis_nan_policy_factory(lambda x: x, n_outputs=1, default_axis=0,
+                          result_to_tuple=lambda x: (x,))
+def _boxcox_llf(data, axis=0, *, lmb):
     xp = array_namespace(data)
     data = xp_promote(data, force_floating=True, xp=xp)
-
-    N = data.shape[0]
+    N = data.shape[axis]
```

Comment:
To add MArray support, this will need to use `_length_nonmasked`, but again, that can be left to a follow-up.